### PR TITLE
add package to handle kill signals

### PR DESCRIPTION
**Pull request by thatInfrastructureGuy**
All the grpc servers  need to add this before they start listening on the port.

```
      import "github.com/TRON-US/go-btfs-common/utils/grpc"
      go func() {
		grpc.GracefulTerminateDetect()
		grpc.GracefulTerminateExec(s)
	}()
```

Post deployment changes:
- [ ]  Status Server
- [ ]  Escrow
- [ ]  Guard Interceptor
- [ ]  Guard Commander
- [ ]  Guard Challenge Preparer

@uchetron will you follow up on implementing the code here if/when this PR is merged and released.
As always we can test with one of the servers.

### CODE REVIEW

**Comment by uchetron** [comment]
LGTM after fmts are changed to logs

File: utils/grpc_graceful_termination.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+// This function only detects the signal. It does not do anything with the detect.
+// Use GracefulTerminateExec func to cleanup else write your own after calling GracefulTerminateDetect
+func (grpcInfo ServerCleanupInfo) GracefulTerminateDetect() {
+	interrupt := make(chan os.Signal, 1)
+	signal.Notify(interrupt, os.Interrupt, syscall.SIGTERM)
+	defer signal.Stop(interrupt)
+
+	select {
+	case <-interrupt:
+		fmt.Println("Received kill signal")
```
[the remark this answers is not in the document]
Comment (thatInfrastructureGuy) [reply utils/grpc_graceful_termination.go]: done

File: utils/grpc_graceful_termination.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+		break
+	}
+}
+
+// GracefulTerminateExec performs a default list of tasks to clean up
+// Override this function in your package to do more.
+func (grpcInfo ServerCleanupInfo) GracefulTerminateExec() {
+	if grpcInfo.GrpcServer != nil {
+		fmt.Println("Shutting down server gracefully")
```
[the remark this answers is not in the document]
Comment (thatInfrastructureGuy) [reply utils/grpc_graceful_termination.go]: done

File: utils/grpc_graceful_termination.go [hunk 3]
```diff
@@ -0,0 +1,40 @@
+package utils
+
+import (
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/tron-us/go-common/v2/log"
```
Comment (wxue) [inline utils/grpc_graceful_termination.go]: need a separate line here between tron-us packages and third-party packages

Comment (thatInfrastructureGuy) [reply utils/grpc_graceful_termination.go]: @wxue Done. 
Also 
1. Converted methods to functions. 
2. Removed struct type as it was not needed. 
3. Graceful Termination timeout has been added. If it takes more than 10 seconds, we stop the server.

Comment (wxue) [reply utils/grpc_graceful_termination.go]: LGTM, thanks.

**Pull request merged**